apps/alert_processor/lib/dissemination/notification_worker.ex

```diff
@@ -37,10 +37,32 @@ defmodule AlertProcessor.NotificationWorker do
         log(id, notification, "event=pop time=#{now() - pop_start}")
 
         send_start = now()
-        NotificationSender.send(notification)
-        log(id, notification, "event=send time=#{now() - send_start}")
 
-        send(self(), :work)
+        case NotificationSender.send(notification) do
+          {:ok, _} ->
+            log(id, notification, "event=send notification=#{id} time=#{now() - send_start}")
+            send(self(), :work)
+
+          {:error, {:http_error, 400, _}} ->
+            SendingQueue.push(notification)
+
+            log(
```

Comment:
suggestion (non-blocking): I don't have strong feelings on this, so take it or leave it, but I wonder if this log line should be a warning and the one below should be an error?

Reply:
As I think about it, the error one seems like a strong case to me since it's a failure that we're not able to properly recover from.

Reply:
Good point, will do.